frontend/src/views/workbench/homePage/utils.ts

```diff
@@ -434,22 +434,15 @@ export function getPieCharOptions(key: WorkCardEnum, hasPermission: boolean) {
 }
 
 // 空数据和无权限处理
-export function handleNoDataDisplay(
-  xAxis: string[],
-  projectCountList: { id: string; name: string; count: number[] }[]
-) {
-  if (!xAxis.length) {
+export function handleNoDataDisplay(xAxis: string[], hasPermission: boolean) {
+  if (!hasPermission) {
     return {
       invisible: false,
       text: t('workbench.homePage.notHasResPermission'),
     };
   }
 
-  const isEmptyData = projectCountList.every((item) =>
-    item.count.every((e) => e === 0 || e === null || e === undefined)
-  );
-
-  if (isEmptyData) {
+  if (!xAxis.length) {
     return {
       invisible: false,
       text: t('workbench.homePage.notHasData'),
```

Comment:
The code for getting bar options with some specific values appears to be correct and clear.

For the pie chart options' handling of missing data and no permission:

`handleNoDataDisplay function`, when it checks "if !hasPermission", should actually check `!hasPermission.value`. This makes sense because is not a typo: `"!!" == true`.
And similarly, you can use this approach in `export function getCommonBarOptions()`.
In case there's any potential change needed based on future changes or additional requirements that may arise during development, one common practice is to include comments explaining your logic. It could also be helpful to format these as inline comments next to your variable names for better readability.
Overall, this code looks good but might need minor adjustments depending on what specifically issues are being faced moving forward.